libs/application/data-providers/src/providers/UserProfileProvider.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  BasicDataProvider,
+  FailedDataProviderResult,
+  SuccessfulDataProviderResult,
+} from '@island.is/application/core'
+
+// TODO using this dataprovider now depends on the developer running the user profile service, and also that (s)he has
+// inserted their mobilenumber AND email accordingly. How can we improve this for local development? Will the
+// shared/mocking lib be good enought for server side mocking?
+export class UserProfileProvider extends BasicDataProvider {
+  readonly type = 'UserProfile'
+
+  async provide(): Promise<unknown> {
+    const query = `query GetUserProfile {
+      getUserProfile {
+        email
+        emailVerified
+        mobilePhoneNumber
+        mobilePhoneNumberVerified
+      }
+    }`
+
+    return this.useGraphqlGateway(query)
```

Comment:
I'm not sure if I love a circular reference between the graphql API and services. For a start, it makes it very tricky to strongly type the requests since the API schemas depend on each other.

Is there an issue with configuring/allowing the application service to depend on other services directly @pshomov?

Reply:
We wanted to try to avoid direct dependencies between services and rather go through the official entry point. Would be much simpler for everyone to reason about dependencies if we know the only way to access a service is through the graphql interface.